src/DataSource/Interpreter/CsvFileInterpreter.php

```diff
@@ -98,13 +112,23 @@ public function previewData(string $path, int $recordNumber = 0, array $mappedCo
                 //load first row and ignore it
                 $data = fgetcsv($handle, 0, $this->delimiter, $this->enclosure, $this->escape);
 
-                foreach ($data as $index => $columnHeader) {
-                    $columns[$index] = trim($columnHeader) . " [$index]";
+                if($this->saveHeaderName){
```

Comment:
Maybe we could put that part of the code into a local, private function for a bit better readability?

Reply:
which code are you referring to? this block is assigning both the `$columns` and setting `$header` in the case that `saveHeaderName` is true.

Reply:
Sorry, I missed something here, we would need to restructure the method as a whole for it to make sense. Thanks for having a look into it anyway!